libraries/chain/database/block_tasks/process_witness_reward_in_sp_migration.cpp

```diff
@@ -0,0 +1,143 @@
+#include <scorum/chain/database/block_tasks/process_witness_reward_in_sp_migration.hpp>
+
+#include <scorum/chain/services/dynamic_global_property.hpp>
+#include <scorum/chain/services/witness_reward_in_sp_migration.hpp>
+#include <scorum/chain/services/account.hpp>
+#include <scorum/chain/services/blocks_story.hpp>
+
+#include <scorum/chain/schema/witness_objects.hpp>
+
+namespace scorum {
+namespace chain {
+namespace database_ns {
+
+const uint32_t process_witness_reward_in_sp_migration::old_reward_alg_switch_reward_block_num = 24;
+const share_type process_witness_reward_in_sp_migration::new_reward_to_migrate = 22831050;
+const share_type process_witness_reward_in_sp_migration::old_reward_to_migrate = 21689497;
+const share_type process_witness_reward_in_sp_migration::migrate_deferred_payment
+    = (process_witness_reward_in_sp_migration::new_reward_to_migrate
+       - process_witness_reward_in_sp_migration::old_reward_to_migrate);
+
+void process_witness_reward_in_sp_migration::on_apply(block_task_context& ctx)
+{
+    dynamic_global_property_service_i& dprops_service = ctx.services().dynamic_global_property_service();
+
+    if (dprops_service.head_block_time() < SCORUM_WITNESS_REWARD_MIGRATION_DATE)
+    {
+        return;
+    }
+
+    witness_reward_in_sp_migration_service_i& witness_reward_in_sp_migration_service
+        = ctx.services().witness_reward_in_sp_migration_service();
+
+    if (!witness_reward_in_sp_migration_service.is_exists())
+    {
+        return;
+    }
+
+    blocks_story_service_i& blocks_story_service = ctx.services().blocks_story_service();
+
+    const auto& dpo = dprops_service.get();
+
+    using recipients_type = std::map<account_name_type, share_type>;
+    uint32_t block_num = dpo.head_block_number;
+    optional<signed_block> b;
+    share_type total_payment = witness_reward_in_sp_migration_service.get().balance;
+    recipients_type witnesses;
+    while (block_num > old_reward_alg_switch_reward_block_num)
+    {
+        b = blocks_story_service.fetch_block_by_number(block_num--);
+        if (b.valid())
+        {
+            signed_block block = *b;
+
+            share_type payment;
+
+            if (total_payment >= migrate_deferred_payment)
```

Comment:
Replace this is-else statement with
`share_type payment = std::min(migrate_deferred_payment, total_payment);`
`total_payment -= payment;`